xpublish/rest.py

```diff
@@ -28,72 +52,74 @@ class Rest:
     Additionally the :class:`xpublish.SingleDatasetRest` class allows has
     a simplified interface for single dataset access.
 
-    Parameters
-    ----------
-    datasets :
-        A mapping of datasets objects to be served. If a mapping is given, keys
-        are used as dataset ids and are converted to strings. See also the notes below.
-    routers :
-        A list of dataset-specific :class:`fastapi.APIRouter` instances to
-        include in the fastAPI application. These routers are in addition
-        to any loaded via plugins.
-        The items of the list may also be tuples with the following format:
-        ``[(router1, {'prefix': '/foo', 'tags': ['foo', 'bar']})]``, where
-        the 1st tuple element is a :class:`fastapi.APIRouter` instance and the
-        2nd element is a dictionary that is used to pass keyword arguments to
-        :meth:`fastapi.FastAPI.include_router`.
-    cache_kws :
-        Dictionary of keyword arguments to be passed to
-        :meth:`cachey.Cache.__init__()`.
-        By default, the cache size is set to 1MB, but this can be changed with
-        ``available_bytes``.
-    app_kws :
-        Dictionary of keyword arguments to be passed to
-        :meth:`fastapi.FastAPI.__init__()`.
-    plugins :
-        Optional dictionary of loaded, configured plugins.
-        Overrides automatic loading of plugins.
-        If no plugins are desired, set to an empty dict.
-
-    Notes
-    -----
-    The urls of the application's API endpoints differ whether a single dataset
-    or a mapping (collection) of datasets is given. In the latter case, all
-    dataset-specific endpoint urls have the prefix ``/datasets/{dataset_id}``,
+    NOTE: The urls of the application's API endpoints differ whether a single
+    dataset or a mapping (collection) of datasets is given. In the latter case,
+    all dataset-specific endpoint urls have the prefix ``/datasets/{dataset_id}``,
     where ``{dataset_id}`` corresponds to the keys of the mapping (converted to
-    strings). Still in the latter case, the endpoint ``/datasets`` is added and returns
-    the list of all dataset ids.
-
+    strings). Still in the latter case, the endpoint ``/datasets`` is added and
+    returns the list of all dataset ids.
     """
 
     def __init__(
         self,
         datasets: Optional[Dict[str, xr.Dataset]] = None,
-        routers: Optional[APIRouter] = None,
+        routers: Optional[Union[APIRouter, List[APIRouter]]] = None,
```

Comment:
Was `routers`, `Optional[APIRouter]` before? If so, that was a mistake, and I think we should simplify the typing to `Optional[List[APIRouter]]` and skip the check below rather than have a `Union`.

Reply:
Works for me. I can implement that.

Reply:
Implemented with commit 69a5a28f58f210ba0460468def23c428ba2133e7